src/core/pointcloud/qgslazdecoder.cpp

```diff
@@ -564,15 +566,19 @@ std::unique_ptr<QgsPointCloudBlock> decompressLaz_( FileType &file, const QgsPoi
     int xAttributeOffset, yAttributeOffset;
     const QgsPointCloudAttribute *attributeX = nullptr;
     const QgsPointCloudAttribute *attributeY = nullptr;
-    const bool hasFilterRect = !filterRect.isEmpty();
-    if ( hasFilterRect )
+    const bool hasFilterGeometry = !filterGeometry.isEmpty();
+    if ( hasFilterGeometry )
     {
       attributeX = requestedAttributes.find( QLatin1String( "X" ), xAttributeOffset );
       attributeY = requestedAttributes.find( QLatin1String( "Y" ), yAttributeOffset );
-      filterRect.setXMinimum( ( filterRect.xMinimum() - offset.x() ) / scale.x() );
-      filterRect.setXMaximum( ( filterRect.xMaximum() - offset.x() ) / scale.x() );
-      filterRect.setYMinimum( ( filterRect.yMinimum() - offset.y() ) / scale.y() );
-      filterRect.setYMaximum( ( filterRect.yMaximum() - offset.y() ) / scale.y() );
+
+      // QgsGeometry::contains return False if a point is at the border of the geometry.
```

Comment:
This is not a good approach. You should be using .intersection here to get the points on the border. And avoid using QgsGeometry::contains (or QgsGeometry::intersection), as it's going to be incredibly expensive to call this for every point. It involves a translation of the geometry to GEOS classes, then calling the non-optimised non-prepared GEOS intersection method. For EVERY SINGLE point fetched. :scream_cat: :scream_cat: :scream_cat: :scream_cat: When testing intersections of more than one object, **always** use prepared GEOS geometries and the optimised prepared checks.

Reply:
This is not needed anymore.